Sources/NIO/CircularBuffer.swift

```diff
@@ -26,62 +26,60 @@ public protocol AppendableCollection: Collection {
 public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
     private var buffer: ContiguousArray<E?>
 
-    /// The capacity of the underlying buffer
-    private var bufferCapacity: Int
-
     /// The index into the buffer of the first item
-    private var startIdx = 0
+    private(set) /* private but tests */ internal var startIdx = 0
```

Comment:
If you're going to expose these as internal I think these variables need different names. They are dangerously close to shadowing the variables `startIndex` and `endIndex` that come from `Collection`, and yet they behave quite differently.

Reply:
Up to @weissi.

Reply:
I like `headIdx` and `tailIdx`, then it's also more obvious that head can be larger than tail